src/kagglehub/models_helpers.py

```diff
@@ -16,11 +17,15 @@ def _create_model(owner_slug: str, model_slug: str) -> None:
     logger.info(f"Model '{model_slug}' Created.")
 
 
-def _create_model_instance(model_handle: ModelHandle, files: List[str], license_name: Optional[str] = None) -> None:
+def _create_model_instance(
+    model_handle: ModelHandle, files_and_directories: UploadDirectoryInfo, license_name: Optional[str] = None
+) -> None:
+    serialized_data = files_and_directories.serialize()
     data = {
         "instanceSlug": model_handle.variation,
         "framework": model_handle.framework,
-        "files": [{"token": file_token} for file_token in files],
+        "files": [{"token": file_token} for file_token in files_and_directories.files],
+        "directories": serialized_data,
```

Comment:
This should be `"directories": [serialized_data],`. Same for the _create_model_instance_version below.

Reply:
Actually, it should be:  `"directories": serialized_data.directories,`